objects/src/block/header.rs

```diff
@@ -275,17 +272,111 @@ impl Deserializable for BlockHeader {
     }
 }
 
-// UTILITIES
+// BLOCK NUMBER
 // ================================================================================================
 
-/// Returns the block number of the epoch block for the given `epoch`.
-pub const fn block_num_from_epoch(epoch: u16) -> u32 {
-    (epoch as u32) << BlockHeader::EPOCH_LENGTH_EXPONENT
+/// A convenience wrapper around a `u32` representing the number of a block.
+///
+/// Each block has a unique number and block numbers increase monotonically by `1`.
+#[derive(Debug, Default, Eq, PartialEq, Copy, Clone, PartialOrd, Ord, Hash)]
+pub struct BlockNumber(u32);
+
+impl BlockNumber {
+    /// The length of an epoch expressed as a power of two. `2^(EPOCH_LENGTH_EXPONENT)` is the
+    /// number of blocks in an epoch.
+    ///
+    /// The epoch of a block can be obtained by shifting the block number to the right by this
+    /// exponent.
+    pub const EPOCH_LENGTH_EXPONENT: u8 = 16;
+
+    /// Returns the previous block number
+    pub fn parent(&mut self) -> u32 {
+        self.0 - 1
+    }
```

Comment:
Here we should probably use `checked_sub` then. If we were to call this on the genesis block (block number `0`) now, it would panic.